ext/standard/password.c

```diff
@@ -61,7 +81,14 @@ static php_password_algo php_password_determine_algo(const char *hash, const siz
 {
 	if (len > 3 && hash[0] == '$' && hash[1] == '2' && hash[2] == 'y' && len == 60) {
 		return PHP_PASSWORD_BCRYPT;
+	} 
+#if HAVE_ARGON2LIB
+	if (hash[0] == '$' && strstr(hash, "argon2i")) {
```

Comment:
The comment about the length check was only to avoid out of bounds memcmp, not ensure a particular value. I.e. something along the lines of:

```c
if (hash_len >= sizeof("$argon2i$")-1 && !memcmp(hash, "$argon2i$", sizeof("$argon2i$")-1)) {
    return PHP_PASSWORD_ARGON2I;
}
// ...
```

You're right that it probably doesn't matter anyway. The only problem I can imagine is the string "argon2i" occurring in the salt of a non-bcrypt non-argon hash (iirc password_verify also accepts crypt hashes that password_hash cannot generate). But this is probably pretty far fetched ^^